fix CI build for Debian

ltalirz opened the pull request:
fixes #38

This PR adds two role dependencies:
 - geerlingguy.docker for docker
 - nginxinc.nginx (the official nginx role)

Both dependencies were necessary to fix the builds on debian 9/10.

Open Questions:
 - I currently add "dokku" as a user who can access the docker daemon without password (hardcoded). Is this necessary / desirable or should the default be empty? 
 - We could make this configurable (in this case one might not even need a dedicated variable in the dokku role and could directly specify the corresponding variable of the docker role - to check)

josegonzalez commented:
- Re: passwordless user, I think thats fine, isn't that how it works by default? Or am I missing something...
- Configurable would be best :)

ltalirz commented:
> Re: passwordless user, I think thats fine, isn't that how it works by default? Or am I missing something...

You are right - on the machine where I installed dokku, I can access the docker daemon without `sudo` as the `dokku` user.
Is that part of the dokku installation?
Because it is not the default behavior in the sense that the `ubuntu` user, through which I ran the ansible role, does not have this privilege.

> Configurable would be best :)

I agree.

josegonzalez commented:
I wonder who owns that socket...

ltalirz commented:
```
$ sudo ls /var/run/docker.sock -ltra
srw-rw---- 1 root docker 0 Jan  3 14:00 /var/run/docker.sock
```
I guess somewhere along the installation, `dokku` gets put into the `docker` group (which is exactly what the geerlongguy.docker role does as well).

josegonzalez commented:
Yeah I think that happens in the dokku installation, since the dokku binary calls out to docker.

ltalirz commented:
I checked that one can specify the `docker_users` variable on the `ansible-dokku` role and that it properly propagates into the `geerlingguy.docker` role.

Link to role variables of docker and nginx now documented in README as well.

josegonzalez commented:
Let me know when you want a review.

ltalirz commented:
> Let me know when you want a review.

Yes, please :-)

I can currently not ask for reviewers via the web api.

josegonzalez commented on README.md:
```diff
@@ -43,29 +43,7 @@ Supported Platforms
-- default: `https://download.docker.com/linux/ubuntu/gpg`
-- type: `string`
-- description: Location for download.docker key
+See also the role variables of the [geerlingguy.docker](https://github.com/geerlingguy/ansible-role-docker) and  [nginxinc.nginx](https://github.com/nginxinc/ansible-role-nginx) role dependencies.
```
The readme gets autogenerated from https://github.com/dokku/ansible-dokku/blob/48fceaea170b872f78de0918b02bb93ee326038e/defaults/main.yml.base and https://github.com/dokku/ansible-dokku/blob/48fceaea170b872f78de0918b02bb93ee326038e/bin/generate so it might make sense to see if you can modify the binary to reference these roles.

ltalirz replied:
Yes - now the roles already show up in the dependency list, so I guess people will figure out they can also set variables for them.

josegonzalez commented on defaults/main.yml:
```diff
@@ -1,8 +1,4 @@
-docker_apt_key_sig: 58118E89F3A912897C070ADBF76221572C52609D
-docker_apt_key_url: hkp://p80.pool.sks-keyservers.net:80
-docker_download_key_sig: 9DC858229FC7DD38854AE2D88D81803C0EBFCD88
-docker_download_key_url: https://download.docker.com/linux/ubuntu/gpg
```
Ditto re: generation.

ltalirz commented:
Done!

josegonzalez commented on Makefile:
```diff
@@ -9,7 +9,7 @@ release: generate
 generate: clean README.md defaults/main.yml
 
-README.md:
+README.md: defaults/main.yml defaults/main.yml.base meta/main.yml
```
Thinking about it, should this be PHONY so we can always regenerate it? Same for anything else that gets auto-generated.

ltalirz replied:
I see. Yes, I'll make it PHONY

josegonzalez commented on README.md:
```diff
@@ -164,7 +145,7 @@ Supported Platforms
 ### sshcommand_version
 
-- default: `0.9.0`
+- default: `0.8.0`
```
I think this got updated in a separate PR and was never updated in the `defaults/main.yml.base` file, mind verifying/fixing if so?

ltalirz replied:
Of course I noticed it only after I pushed.
Those last-minute pushes before leaving the office ... ;-)

josegonzalez commented on defaults/main.yml:
```diff
@@ -15,4 +11,4 @@ dokku_vhost_enable: 'true'
 herokuish_version: 0.5.5
 plugn_version: 0.3.2
-sshcommand_version: 0.9.0
+sshcommand_version: 0.8.0
```
Yeah looks like we didn't update `defaults/main.yml.base`. Mind fixing there and regenerating?

josegonzalez replied:
Can you update the valie in `defaults/main.yml.base` to be `0.9.0` and regenerate? Sorry if that was not clear.

josegonzalez commented on bin/generate:
```diff
@@ -124,7 +124,7 @@ def add_requirements(text, meta, defaults):
     dependencies = []
     if len(meta['dependencies']) > 0:
-        dependencies = ['- {0}'.format(dependency) for dependency in defaults['dependencies']]
+        dependencies = ['- {0}'.format(dependency) for dependency in meta['dependencies']]
```
Can we append ` ansible role` to the string here so that the readme has the right text?

ltalirz replied:
Sure.

josegonzalez commented on tasks/install.yml:
```diff
@@ -1,58 +1,15 @@
   apt_repository:
     filename: dokku
-    repo: 'deb https://packagecloud.io/dokku/dokku/ubuntu/ {{ ansible_lsb.codename|lower }} main'
+    repo: 'deb https://packagecloud.io/dokku/dokku//{{ ansible_distribution|lower }} {{ ansible_distribution_release }} main'  # yamllint disable-line rule:line-length
```
I think the trailing slash was changed incorrectly here.

ltalirz replied:
Thanks for spotting.
Interesting that this still seemed to work...

ltalirz commented:
I also removed the `defaults/main.yml` target in the `Makefile` since I believe it was superfluous (?)

I've cleaned up the last few commits a bit - still, feel free to "squash & merge" when it's ready.

josegonzalez commented on Makefile:
```diff
@@ -7,10 +7,8 @@ release: generate
 README.md:
 	bin/generate
-
-defaults/main.yml:
```
I think we still want this, and it should mirror the README.md command.

ltalirz replied:
makes sense  - I overlooked that it also touches the yml file

josegonzalez commented:
Still a few changes left, but close!

josegonzalez commented on .travis.yml:
```diff
@@ -26,6 +26,15 @@ before_script:
+      git status --short;
+      git diff;
+      exit 1;
+    fi
```
👏

ltalirz commented:
Looks like the tests pass ;-)
Sorry for the back & forth on this one
